### PR TITLE
[Discover] options button in canvas to toggle legacy mode

### PR DESCRIPTION
**Pull request by kavilla**
### Description

Removes top nav bar link and provides a button option.

### Issues Resolved

n/a

## Screenshot

Screenshot 2024-03-16 at 12 23 07 AM

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by ashwin-pc — APPROVED** [review]

File: src/plugins/discover/public/application/view_components/canvas/index.tsx [hunk 1]
```diff
@@ -96,6 +97,50 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history }: ViewPro
     }
   };
 
+  const [isOptionsOpen, setOptionsOpen] = useState(false);
+  const [useLegacy, setUseLegacy] = useState(!getNewDiscoverSetting(storage));
+  const DiscoverOptions = () => (
```
Comment (ruanyl) [inline src/plugins/discover/public/application/view_components/canvas/index.tsx:102]: Could consider to move `DiscoverOptions` out of `DiscoverCanvas` for code readability and  reusability

**Review by ruanyl — APPROVED** [review]

**Comment by kavilla** [comment]
@kgcreative what do you think about this?

**Comment by kgcreative** [comment]
@kavilla If we're able to align the button top-right instead of top-left, i think that's more consistent with other places where we do additional actions, but that's a minor nice-to-have

**Comment by kavilla** [comment]
> @kavilla If we're able to align the button top-right instead of top-left, i think that's more consistent with other places where we do additional actions, but that's a minor nice-to-have

like this:
Screenshot 2024-03-18 at 12 58 03 PM

**Comment by kgcreative** [comment]
> > @kavilla If we're able to align the button top-right instead of top-left, i think that's more consistent with other places where we do additional actions, but that's a minor nice-to-have
> 
> like this @kgcreative?: Screenshot 2024-03-18 at 12 58 03 PM

yes, that's perfect.

**Review by abbyhu2000 — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

File: src/plugins/discover/public/application/view_components/canvas/index.tsx (hunk 1 above)
Comment (kavilla) [reply src/plugins/discover/public/application/view_components/canvas/index.tsx:102]: good point. ideally, there is a shelf life of this. until we match performance and feature parity of the legacy and current Discover

**Review by ruanyl — APPROVED** [review]

**Pull request merged**